typing_extensions/src_py2/typing_extensions.py

```diff
@@ -0,0 +1,113 @@
+import abc
+import collections
+import typing
+from typing import (
+    ClassVar, Type,
+    Counter, DefaultDict, Deque,
+    NewType, overload, Text, TYPE_CHECKING,
+)
+
+# Please keep __all__ alphabetized within each category.
+__all__ = [
+    # Super-special typing primitives.
+    'ClassVar',
+    'Type',
+
+    # Concrete collection types.
+    'Collection',
+    'ContextManager',
+    'Counter',
+    'Deque',
+    'DefaultDict',
+
+    # One-off things.
+    'NewType',
+    'overload',
+    'Text',
+    'TYPE_CHECKING',
```

Comment:
I'm not sure if we should -- GenericMeta seems to have gone through some heavy changes since 3.5.0, and if we make it public here, we're basically guaranteeing it'll work reasonably consistently across all Python versions, and I'm not sure if it'll be easy for us to confirm that's really the case.

If you think it's still worth including though, I can add some tests for it + add it to `__all__`.